sequencer/src/bin/espresso-dev-node.rs

```diff
@@ -259,6 +260,19 @@ async fn main() -> anyhow::Result<()> {
             .get_contract_address(Contract::LightClientProxy)
             .unwrap();
 
+        // confirm that the light_client_address is a proxy
+        // using the implementation slot, 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc, which is the keccak-256 hash of "eip1967.proxy.implementation" subtracted by 1
+        let hex_bytes =
```

Comment:
Could we extract this check into a function to re-use the code? We could add  method to the L1 client with this check.

Currently it's not quite clear where would be a good place to perform the check so that if one starts up a local demo or tries to apply an upgrade with a bad fee contract address the user would see an error.

Maybe we could check here

https://github.com/EspressoSystems/espresso-sequencer/blob/31cd49fde6530c71b98197fbb86cc033c4101738/types/src/v0/impls/state.rs#L515

Or maybe there is a better place? 

cc @imabdulbasit

Reply:
I think that we can add this check to Genesis struct maybe and call it when the node starts?

Reply:
so we panic on the startup

Reply:
yes

Reply:
in future we can rename this to `validate()` and validate other things including the contract

Reply:
Just to clarify, I don't see light client contract address anywhere for sequencer main binary. We only have fee contract address in genesis toml file. So, I think we should validate this in separate function and just call it in sequencer main binary

Reply:
Yeah the LC contract is used by the prove, so we should check the LC contract when starting up the prover.

Reply:
where should the LC check be? here? https://github.com/EspressoSystems/espresso-sequencer/blob/1527-sequencer-nodes-should-do-a-sanity-checks-to-check-if-the-l1-contract-addresses-are-wrong/hotshot-state-prover/src/bin/state-prover.rs#L98

Reply:
the check for the fee contract in sequencer main is here  https://github.com/EspressoSystems/espresso-sequencer/blob/1527-sequencer-nodes-should-do-a-sanity-checks-to-check-if-the-l1-contract-addresses-are-wrong/sequencer/src/main.rs#L28 and the implementation of that contract is here https://github.com/EspressoSystems/espresso-sequencer/blob/1527-sequencer-nodes-should-do-a-sanity-checks-to-check-if-the-l1-contract-addresses-are-wrong/sequencer/src/genesis.rs#L80https://github.com/EspressoSystems/espresso-sequencer/blob/1527-sequencer-nodes-should-do-a-sanity-checks-to-check-if-the-l1-contract-addresses-are-wrong/sequencer/src/genesis.rs#L80

Reply:
made the check for the light client contract [here](https://github.com/EspressoSystems/espresso-sequencer/pull/2000/commits/3c7614d7c0047d40a007cdab0f1ddde9e7497cb6#diff-53f5d56b703d684326ffd2a4011ae3fcf58e68deb773b916efd4028f52b4c1c5R114) and the implementation of the verification is [here](https://github.com/EspressoSystems/espresso-sequencer/pull/2000/commits/3c7614d7c0047d40a007cdab0f1ddde9e7497cb6#diff-8bcfa1727c03a3fa83a1b636f2ba693ff0dc52770828f180dabf4b541236c0e3R85).